Docs: Adds zero+few shot tutorial with SetFit

dvsrepo opened the pull request:
# Description

Includes a new tutorial for Zero-shot with SetFit

dvsrepo committed:
Docs: Adds zero+few shot tutorial with SetFit

dvsrepo committed:
chore: fix typo

tomaarsen commented on docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb:
```diff
@@ -0,0 +1,645 @@
+      "source": [
+        "## 🔫 Zero-shot predictions with SetFit\n",
+        "\n",
+        "The original `banking77` dataset is a intent classification dataset with dozens of labels (`lost_card`, `card_arrival`, etc.). To keep this tutorial simple, we define a simplified labelling scheme with higher level classes.\n",
```
```suggestion
        "The original `banking77` dataset is an intent classification dataset with dozens of labels (`lost_card`, `card_arrival`, etc.). To keep this tutorial simple, we define a simplified labelling scheme with higher level classes.\n",
```

tomaarsen commented on docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb:
```diff
@@ -0,0 +1,645 @@
+        "id": "Sp3oSMi9snkp"
+      },
+      "source": [
+        "Let's use our trained model to predict over the dataset. We'll later load this predictions into Argilla and use them to speed-up the labelling process."
```
```suggestion
        "We can use our trained zero-shot model to predict over the dataset. We'll later load these predictions into Argilla and use them to speed-up the labelling process."
```
This one is a bit nitpicky. I think it's good to include `zero-shot` for clarity, and `Let's` was used 3 times in a row, so I changed it up too. `this` was incorrect.

tomaarsen commented on docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb:
```diff
@@ -0,0 +1,645 @@
+        "3. You can now review the predictions, validate them, or correct them.\n",
+        "\n",
+        "\n",
+        "After labeling around 200 records, we're ready to evaluate our zero-shot model, let's see how! "
```
```suggestion
        "After labelling around 200 records, we're ready to evaluate our zero-shot model, let's see how! "
```
Either "labeling" or "labelling" is fine, but we use them both in the same tutorial now, which isn't great. Because we use `labelling` more, I'll go with that one. (But feel free to go with `labeling` instead)

dvsrepo committed:
chore: change title

tomaarsen commented on docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb:
```diff
@@ -0,0 +1,645 @@
+      "source": [
+        "## 📏 Evaluate the zero-shot model\n",
+        "\n",
+        "We can use Argilla's built-in metrics to compute `f1` based on (1) the predictions of the zero-shot model we stored at the beginning of the tutorial, and (2) the manual annotations. Please note that during the labeling process, we've added a new label `Other` to account for examples that didn't fall into our predefined categories. This highlights the importance of iterating early on duri ...
```
```suggestion
        "We can use Argilla's built-in metrics to compute `f1` based on (1) the predictions of the zero-shot model we stored at the beginning of the tutorial, and (2) the manual annotations. Please note that during the labelling process, we've added a new label `Other` to account for examples that didn't fall into our predefined categories. This highlights the importance of iterating early on during project definition. Argilla gives users a lot of flexibility, features like predictions and similarity search can help to surface potential issues and refinements much faster than with traditional data annotation tools. "
```
Same as above, also put a comma in a long sentence

dvsrepo committed:
Update docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb

Co-authored-by: Tom Aarsen <37621491+tomaarsen@users.noreply.github.com>

dvsrepo committed:
Update docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb

Co-authored-by: Tom Aarsen <37621491+tomaarsen@users.noreply.github.com>

dvsrepo committed:
Update docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb

Co-authored-by: Tom Aarsen <37621491+tomaarsen@users.noreply.github.com>

dvsrepo committed:
Update docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb

Co-authored-by: Tom Aarsen <37621491+tomaarsen@users.noreply.github.com>

tomaarsen commented on docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb:
```diff
@@ -0,0 +1,645 @@
+        "id": "Y-e_l9dtXywV"
+      },
+      "source": [
+        "## 🏷 Bulk labelling with the `find similar` and zero-shot predictions\n",
```
```suggestion
        "## 🏷 Bulk labelling with `find similar` and zero-shot predictions\n",
```
I think this is slightly better? Not 100% on this one. The original isn't too bad either way.

tomaarsen commented on docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb:
```diff
@@ -0,0 +1,645 @@
+      },
+      "source": [
+        "## 🏷 Bulk labelling with the `find similar` and zero-shot predictions\n",
+        "Now that our `banking77-topics-setfit` is available from the Argilla UI. You can start annotating the data leveraging similarity search and zero-shot predictions. After going to your Argilla UI URL, the workflow is following:\n",
```
```suggestion
        "Now that our `banking77-topics-setfit` is available from the Argilla UI. You can start annotating the data by leveraging similarity search and our zero-shot predictions. After going to your Argilla UI URL, the workflow is following:\n",
```

tomaarsen commented on docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb:
```diff
@@ -0,0 +1,645 @@
+        "id": "6JFCGtGlsh0x",
+        "outputId": "911e0115-97a1-4a30-ec3b-1fa963a4f26c"
+      },
+      "outputs": [
```
This cell its output does not want to show: https://docs.argilla.io/en/docs-adds_zeroshot_setfit_tutorial/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.html

dvsrepo replied:
I think it was caused by exporting from Colab. I tried now to rebuild the output locally now and we'll see if it renders

tomaarsen commented on docs/_source/tutorials/notebooks/labelling_textclassification_setfit_zeroshot.ipynb:
```diff
@@ -0,0 +1,645 @@
+        "# Load the handlabelled dataset from Argilla\n",
+        "ds = rg.load(\"banking77-topics-setfit\").prepare_for_training(train_size=0.8)\n",
+        "\n",
+        "# Load SetFit model from Hub\n",
```
```suggestion
        "# Load SetFit model using a Sentence Transformer from the Hub\n",
        "# Feel free to experiment with other models, e.g. \"sentence-transformers/paraphrase-mpnet-base-v2\"\n",
```
The second comment is optional. The all-MiniLM model is ~90MB while the paraphrase is ~410MB. On some datasets, the paraphrase one outperforms the other by like 15% accuracy. I tend to use the paraphrase model for my personal tests.

dvsrepo replied:
Yes, I agree. I think this suggestion it's a good compromise. I use the all-MiniLM for tutorials because we want them to run very fast even with CPU and it gives decent results for the kind of quickstart approaches we show. That said, all tutorials will benefit from clearer tips.

Done

tomaarsen commented on docs/_source/_static/tutorials/labelling_textclassification_setfit_zeroshot/modal.md:
````diff
@@ -0,0 +1,8 @@
+```{grid-item-card}  🔫 Zero-shot and few-shot classification with SetFit
+:img-top: /_static/tutorials/labelling_textclassification_setfit_zeroshot/labelling_textclassification_setfit_zeroshot.png
+:link: ../../tutorials/notebooks/labelling_textclassification_setfit_zeroshot.html
+MLOps Steps: Training \
````
```suggestion
MLOps Steps: Labelling, Training \
```
I'm guessing this tutorial falls under Labelling too.

tomaarsen commented on docs/_source/tutorials/steps/1_labelling.md:
````diff
@@ -5,6 +5,10 @@
+```{include} /_static/tutorials/labelling_textclassification_setfit_zeroshot/modal.md
+```
+```{include} /_static/tutorials/labelling_textclassification_setfit_zeroshot/modal.md
+```
````
This is double. Perhaps one was meant to be the few-shot tutorial?

tomaarsen committed:
Add self-referential links between SetFit tutorials

tomaarsen committed:
Rename 'labelling_text..." to "labelling-text"

To match the style of all other tutorials

tomaarsen committed:
Also updated the name in the files

I forgot to accept my mass-replace command haha

tomaarsen committed:
Use - for lists in markdown rather than *

tomaarsen committed:
Add additional space before list

Wow, the doc builder is very specific about this, huh. It renders fine in Markdown

dvsrepo committed:
Fix graph and address comments

dvsrepo committed:
Delete =4.2.0

dvsrepo committed:
fix file names

dvsrepo committed:
Comment from Tom about paraphrase mpnet

dvsrepo committed:
Adress comment by tom

dvsrepo committed:
try to fix plotly graph htlm rendering

dvsrepo committed:
trying again

dvsrepo committed:
build with jupyterlab

dvsrepo committed:
address final tom's comments

dvsrepo committed:
final changes

dvsrepo committed:
try to fix summary bullet points

tomaarsen reviewed (approved):
That works! I think that we're ready.
The failing CIs can be ignored.
